instrumentation/netty/netty-4-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/netty/common/AbstractNettyChannelPipelineInstrumentation.java

```diff
@@ -125,6 +142,11 @@ public static void removeHandler(
       if (ourHandler != null) {
         pipeline.remove(ourHandler);
         contextStore.put(handler, null);
+      } else if (handler
+          .getClass()
+          .getName()
+          .startsWith("io.opentelemetry.javaagent.instrumentation.netty.")) {
+        pipeline.removeLast();
```

Comment:
Hmm, what if somebody calls `pipeline.addAfter(originalHandler, ...)`? Then we'll remove some random custom handler that has nothing to do with our instrumentation. If we want this method to work correctly we should
1. Maintain a two-way association between the `originalHandler` and `ourHandler` (i.e. put `ourHandler -> originalHandler` into the `ContextStore` too) and remove the `originalHandler` whenever `removeLast()` removes our handler
2. Instrument `addAfter()` so that it actually adds the new handler after `ourHandler` whenever it's associated to the `originalHandler`

Reply:
yes, u r right.Maybe we can instrument `addAfter()` to make sure  `ourhandler` will always be added after the `originalHandler`.  I'll have a try.